### PR TITLE
Improve Bandcamp originUrl

### LINKED ISSUE #3251: Incorrect originUrl when scrobbling on the Bandcamp home page
**Describe the bug**

**How to reproduce**
1. Go to the [Bandcamp homepage](https://bandcamp.com/)
2. Click to play any playable element on the page
3. See the `originUrl` in the scrobble

**Expected behavior**
The originUrl should be a URL to the album being played rather than the generic homepage URL.

**Debug logs**
See the generic `originUrl`:
```
Web Scrobbler: {
  "track": "March of the Dead (Intro feat. John Carpenter and Cody Carpenter)",
  "artist": "DANCE WITH THE DEAD",
  "album": "Driven to Madness",
  "albumArtist": null,
  "uniqueID": null,
  "duration": null,
  "currentTime": null,
  "isPlaying": true,
  "trackArt": "https://f4.bcbits.com/img/a1386566724_2.jpg",
  "isPodcast": false,
  "originUrl": "https://bandcamp.com/?from=menubar_logo_logged_in"
}
```

**Environment (please complete the following information)**
-   OS:  Mac OS
-   Browser: Brave
-   Extension version: 2.62.0

**Additional context**
There are two types of playable elements on the Bandcamp home page.
One type is tied to a music player on the page while the other is not. So using the music player to get the album URL won't work for both.
However, the DOM elements all get the `.playing` css class added when they are being played (so that they can show a play/pause icon over the album art). This allows us to target whatever the currently playing item might be.

Albums in the "New and notable" section can be targeted with `document.querySelector('.playing + .item-title a').href` (note the  anchor element is a child of the `item-title` element):
image

On the other hand, in the "discover" section at the bottom of the page, the album links can be targeted with `document.querySelector('.playing + .item-title').href` (note the  `item-title` element itself is the anchor element).
image

### PR DESCRIPTION
**Pull request by inverse**
Resolves #3251  #3249 #3246

@MonkeyDo thanks for raising the tickets - this should fix two of the three!

### CODE REVIEW

File: src/connectors/bandcamp.js [hunk 1]
```diff
@@ -213,6 +207,9 @@ function initPropertiesForHomePage() {
 
 		return null;
 	};
+
+
+	Connector.getOriginUrl = () => getOriginUrl('.playable.playing');
```
[the remark this answers is not in the document]
Comment (inverse) [reply src/connectors/bandcamp.js]: I applied the change - let me know how it works?

image

 noticed this section doesn't respond to changes once initially played - but I think that's another issue unrelated to this.

File: src/connectors/bandcamp.js [hunk 2]
```diff
@@ -139,15 +139,7 @@ function initPropertiesForCollectionsPlayer() {
...
-	Connector.getOriginUrl = () => {
-		const buyNowAnchor = document.querySelector('.buy-now a');
-		if (buyNowAnchor === null) {
-			Util.debugLog('Failed to resolve originUrl');
-			return document.location.href;
-		}
-
-		return buyNowAnchor.href.split('?')[0];
-	};
+	Connector.getOriginUrl = () => getOriginUrl('.buy-now a');
```
Comment (MonkeyDo) [inline src/connectors/bandcamp.js:142]: This should cover the 'collections' tab as well as the 'wishlist' tab (which are on the same page)

```suggestion
	Connector.getOriginUrl = () => getOriginUrl('.playing .collection-title-details a, .playing .buy-now a');
```

**Review by MonkeyDo — COMMENTED** [review]
I've tested the PR locally, scrobbling on the home page now works as expected (apart from, as you said, an unrelated issue with the "new and notable" section).
So that's issue #3251 fixed!

One other issue now fixed with this PR is #3246 which isn't in the PR description.
Feed page working fine !

Regarding issue #3249, we have a similar situation to the home page where there are two different types of elements a user can play: one on the "collections" tab and one on the "wishlist" tab. (changing tabs doesn't trigger a page load so detecting which tab we're on would be tricky; currently the `isCollectionsPage` method returns true for either tab).

Considering the fallback selector idea seems to be working nicely, I'm suggesting another double-selector below to fix the wishlist tab.

Thanks again :)

File: src/connectors/bandcamp.js (hunk 2 above)
Comment (inverse) [reply src/connectors/bandcamp.js:142]: Looks like we can even simplify it

Comment (inverse) [reply src/connectors/bandcamp.js]: Actually it didn't work and applied your full fix - thanks :pray:

**Comment by MonkeyDo** [comment]
Thank you so much !

**Pull request merged**